src/views/Permissions.vue

```diff
@@ -0,0 +1,301 @@
+              <ion-icon :icon="openOutline" slot="end" />
+            </ion-button>
+          </div>
+          <hr />
+          <PermissionItems />
+        </main>
+        <main v-else class="empty-state">
+          <p>{{ "Select a security group to view its details" }}</p>
+        </main>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script lang="ts">
+import {
+  IonButton,
+  IonButtons,
+  IonContent,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonNote,
+  IonPage,
+  IonTitle,
+  IonToolbar,
+  alertController
+} from '@ionic/vue';
+import { defineComponent } from 'vue';
+import { translate } from '@hotwax/dxp-components';
+import { addOutline, downloadOutline, idCardOutline, openOutline } from 'ionicons/icons';
+import { useRouter } from 'vue-router';
+import PermissionItems from '@/components/PermissionItems.vue'
+import { mapGetters, useStore } from 'vuex';
+import { jsonToCsv, showToast } from '@/utils';
+import { hasError } from '@/adapter';
+import { UtilService } from '@/services/UtilService';
+import emitter from "@/event-bus";
+import { PermissionService } from '@/services/PermissionService';
+import { DateTime } from 'luxon';
+
+export default defineComponent({
+  name: 'Permissions',
+  components: {
+    IonButton,
+    IonButtons,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonNote,
+    IonPage,
+    IonTitle,
+    IonToolbar,
+    PermissionItems
+  },
+  data() {
+    return {
+      securityGroupUsers: {} as any
+    }
+  },
+  computed: {
+    ...mapGetters({
+      securityGroups: 'util/getSecurityGroups',
+      permissionsByGroupType: 'permission/getPermissionsByGroupType',
+      currentGroupPermissions: 'permission/getCurrentGroupPermissions',
+      currentGroup: 'permission/getCurrentGroup',
+      allPermissions: 'permission/getAllPermissions'
+    })
+  },
+  async mounted() {
+    await this.store.dispatch('util/getSecurityGroups')
+    if(!this.allPermissions.length) await this.store.dispatch('permission/getAllPermissions')
+    if(!Object.keys(this.permissionsByGroupType).length) await this.store.dispatch('permission/getPermissionsByGroupType')
+    if(this.currentGroup.groupId) await this.store.dispatch('permission/getPermissionsByGroup', this.currentGroup.groupId)
+    await this.store.dispatch('permission/checkAssociated')
+  },
+  methods: {
+    createGroup() {
+      this.$router.replace({ path: `/create-security-group/` })
+    },
+    async updateCurrentGroup(group: any) {
+      emitter.emit('presentLoader')
+      await this.store.dispatch('permission/updateCurrentGroup', group)
+      await this.store.dispatch('permission/getPermissionsByGroup', this.currentGroup.groupId)
+      await this.store.dispatch('permission/checkAssociated')
+      await this.getUsersCount()
+      emitter.emit('dismissLoader')
```

Comment:
Check if this loader is dismiss correctly, when we have all the data available in the app and we do not make any api call.